### PR TITLE
Add linkml semantic schemas

### PR DESCRIPTION
**Pull request by paul121**
Standards-8: https://regennetwork.atlassian.net/jira/software/c/projects/ENG/boards/51?selectedIssue=STANDARDS-8

This takes some first steps at including LinkML schemas in this repo. I've only added the existing Project and ProjectPost linkml schemas I've worked on and didn't modify them in this process.

Overview of included changes:

For Schemas:
- Add a `schema` directory at root of this repo
- Provide linkml schemas in `*.yaml` format
- Use linkml [`gen-doc`](https://linkml.io/linkml/generators/markdown.html) utility to generate markdown files for the schema classes, slots (attributes) and data types. I have committed these generated markdown files (~50 total) here but this could potentially happen during a CI/CD build step?

For the Astro site:
- Add a `schema/[...slug]` dynamic route that loads the generated markdown files and renders them in an existing layout template. This is actually quite easy thanks to the [`Astro.glob()`](https://docs.astro.build/en/reference/api-reference/#astroglob) command which can load Markdown files from any directory and prepare them for rendering in a standard Astro component.
 - Add a `Schemas` item to the sidebar, right now just hard-code the top-level schema classes that are important, eg; `Project` and `Project Post`.
 - I started making some follow up changes to improve the sidebar (collapse by default) and change the taxonomy terms to have a `/taxonomy/{taxonomy}/{term}` path.
 
Things to review & some notes:
- For the schemas, I hope this demonstrates a simple first step at how this can be expanded to include core credit class, credit batch, etc types. Any Q's on how this would work?
- There are more schema specific features and options we could implement:
  - adding more descriptions (should do this)
  - providing example datasets (these examples would be included in generated docs)
  - include links to GH source LinkML schemas
- For integration with Astro, what do we think of this approach?
  - I considered trying to use astro's 'content collections' concept that is already implemented for taxonomy terms. In theory we could dump generated markdown into `src/content/schemas` but one downside is that the generated markdown doesn't include any frontmatter that these content collections expect you to have. It seems that we can provide jinja2 templates for the `gen-doc` command where we could potentially include frontmatter... but still, it's more work, and I think we want the schema markdown to follow a different editing flow than taxonomy terms, so it might be better to not use 'content collections'.
- I'm curious the latest thoughts on renaming this repo/site to "Framework" - it seems this might be in-line with the latest thinking on Registry 2.0 I'm just learning more about recently? IMO renaming to 'framework' makes a lot of sense! But with this, I'm curious the goals & types of users we want this page to serve. It's more technical, but how technical is too much? cc @beccaRND @S4mmyb 
- And re: URL paths and term names. I know there was discussion on this in Slack on this awhile back, but I don't think consensus was reached. Assuming we do want a single site that includes schemas, taxonomies, standards, etc. I think breaking down this site navigation to create some separation between each of these concepts will be necessary and thus would make sense to parallel this navigation in the URL paths. I don't see how 'longer urls' create any issues with how these things are used.

### CODE REVIEW

File: src/pages/schema/[...slug].astro
Comment (paul121) [inline src/pages/schema/[...slug].astro:17]: Currently these paths include the `.md` extension, eg `/schema/Project.md`. We can remove the file extension but then the relative hyperlinks from the generated markdown files don't work, since they include `.md` extensions. Two potential solutions: 1) modify hyperlinks in markdown files (I don't think there is out-of-the-box support for this) or 2) make redirects in the Astro site to handle `*.md` extensions`?

**Comment by paul121** [comment]
Some screenshots of what this looks like:

## Root schemas page (generated by linkml)

Screenshot from 2024-07-27 13-47-41

## Schema class page w/ mermaid ER diagram
Screenshot from 2024-07-27 13-48-16

## Schema class page w/ slots (attributes) & hyperlinks to each slot and data type
Screenshot from 2024-07-27 13-48-39

**Comment by paul121** [comment]
> But have a hunch that this might be due to the [rehype-mermaid](https://github.com/remcohaszing/rehype-mermaid) dependency which requires [playwright](https://playwright.dev/) and an additional chromium dependency,

Hmm seems like there is another issue. Will need to check the logs. cc @clevinson

**Review by clevinson — APPROVED** [review]

**Comment by clevinson** [comment]
2 follows we discussed on the Framework WG call today, which we should tackle after merging this:
- Nest "Impacts", "Activities", "Environment Types" under a "Terminology" dropdown on the sidebar
- Explore ways of editing the linkml generated documentation to either clarify or remove language like "Slot" from the docs

@aaronc @paul121 @S4mmyb Anything else we had as a follow up?

**Review by aaronc — COMMENTED** [review]
All the slot links are broken. For example: https://paul-schemas.regen-data-standards.pages.dev/schema/Project.md/description.md. Also, `Project.md/description.md` is not a good URL format.

**Comment by paul121** [comment]
> All the slot links are broken. 

Good catch. It seems this is because all pages are redirecting to an absolute path and the links in the markdown/html are relative. In my local environment I can load a page without the absolute path and it works fine. I saw there is an astrojs config option for [`trailingSlash`](https://docs.astro.build/en/reference/configuration-reference/#trailingslash) but it doesn't seem like Cloudflare is respecting this. I wonder if the Cloudflare project has something configured to have an absolute path?

> Also, Project.md/description.md is not a good URL format.

Agreed. See my inline comment above. I think there are a few ways we could remove the `.md` extension. I just found that astro has a `redirects` config that works but the UX isn't that great on Cloudlfare because it loads a blank page. Locally this redirect was working much smoother.

**Comment by paul121** [comment]
Okay, I solved the issue of slot links not working. We can use the Astro `file` [build format](https://docs.astro.build/en/reference/configuration-reference/#buildformat).

It turns out that cloudflare's [route matching](https://developers.cloudflare.com/pages/configuration/serving-pages/#route-matching) behavior cannot be configured:
> /contact.html will be redirected to /contact, and /about/index.html will be redirected to /about/.

By default Astro uses the `directory` build format and generates a directory with a nested html file for each page, `/schema/Project/index.html`. This explains why we always get absolute URLs in Cloudflare but not locally. Using the `file` build format generates an html file named for each page, `/schema/Project.html`, and should work exactly how we need.

----

This leaves the site functional but with an annoying redirect from `/schema/slug.md` -> `/schema/slug`. I think this is just a limitation of using [redirects](https://docs.astro.build/en/guides/routing/#redirects) on an astro static site:
> Astro will output HTML files with the [meta refresh](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#examples) tag by default.

@aaronc to solve this redirect issue I think we need to add a preprocess or build step that removes the `.md` extension from links in the LinkML generated Markdown files. This way we don't need the complexity of redirects, we get correct links from the get-go. I don't think LinkML has an option to configure how these links are generated in the markdown but it could be a fairly easy option we contribute to LinkML. Otherwise, I think there are a few ways we could pre-process the markdown in JS as a part of the `astro build` process. Either way I'm not sure it needs to block this PR right now, curious what you think!

**Review by aaronc — APPROVED** [review]
Just discussed with @paul121. Let's merge this but in follow ups work on the Astro routing of this and decide whether we want to continue with Astro.glob or use content collections for the schema pages, so that the URL behavior is better.

**Review by clevinson — APPROVED** [review]

**Pull request merged**
